lib/view_component/with_content_helper.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module WithContentHelper
+    def with_content(value = nil)
+      if value.nil?
+        raise ArgumentError.new("No content provided.")
+      else
+        @_content_set_by_with_content = -> (_component) { value }
```

Comment:
Why are we converting this to a proc vs just storing the value?

Reply:
good catch! This was done because we used to accept `blocks` here, so I wanted to keep the interface instead of always checking if `@_content_set_by_with_content` is callable or not